src/error.rs

```diff
@@ -75,6 +94,31 @@ impl ApiError {
         }
         false
     }
+
+    fn weave_error_code(&self) -> WeaveError {
+        match self.kind() {
+            ApiErrorKind::Validation(ver) => match ver.kind() {
+                ValidationErrorKind::FromDetails(ref description, ref location, name) => {
+                    let name = name.clone().unwrap_or("".to_owned());
+                    if *location == RequestErrorLocation::Body
+                        && ["bso", "bsos"].contains(&name.as_str())
+                    {
+                        if description.starts_with("Invalid JSON") {
+                            // Python's tests feel differently about when its invalid JSON
+                            // In Python's case, its a JSON deserialization failure, not merely
+                            // invalid JSON per the desired output.
+                            return WeaveError::InvalidWbo;
```

Comment:
```suggestion
                            return WeaveError::MalformedJson;
```

Reply:
While this would make sense, it would not match the Python tests, which consider this to be an Invalid Wbo due to the reason I doc'd here.

Reply:
I think I misunderstood the comment then. do we even need the block then? it returns InvalidWbo in both cases